### PR TITLE
feat: Adding Webhooks to Index Rate Alert

### PR DESCRIPTION
**Pull request by arani-sen**
adding the ability to add webhooks for
Index Rate Alerts

Ref:[LOG-9667](https://mezmo.atlassian.net/browse/LOG-9667)

[LOG-9667]: https://mezmo.atlassian.net/browse/LOG-9667?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: logdna/resource_alert_test.go
Comment (seeruyy) [inline logdna/resource_alert_test.go:3]: This file has no changes, it has to be discarded

File: logdna/resource_index_rate_alert.go
Comment (seeruyy) [inline logdna/resource_index_rate_alert.go:7]: Formatting

**Review by seeruyy — CHANGES_REQUESTED** [review]

**Review by jsumners — COMMENTED** [review]
I believe we should get https://github.com/logdna/terraform-provider-logdna/pull/88 merged first so that you can run `go fmt ./...` on your branch and see minimal changes. Or, we can ignore my comments about formatting and I'll rebase my PR after this one has merged.

But before either can be merged we need @jakedipity to answer a question I have put to him in Slack about the failing tests.

File: logdna/response_types.go
Comment (seeruyy) [inline logdna/response_types.go:116]: I'd implement this as a regular function, the name of this function does not match with other functions 
```js
func mapIndexRateAlertWebhookToSchema(indexRateAlert indexRateAlertResponse) ([]interface{}) {
	webhooks := make([]interface{}, 0)

	for _, webhook := range indexRateAlert.Channels.Webhook {
		w := make(map[string]interface{})

		w["bodytemplate"] = webhook.BodyTemplate
		w["headers"]      = webhook.Headers
		w["method"]       = webhook.Method
		w["url"]          = webhook.URL

		webhooks = append(webhooks, w)
	}
	return webhooks
}
```

File: docs/resources/logdna_index_rate_alert.md
Comment (seeruyy) [reply docs/resources/logdna_index_rate_alert.md:27]: Don't see fixes added for this comment. This is important documentation which will be posted on a customer used website.
https://registry.terraform.io/providers/logdna/logdna/latest/docs/resources/logdna_index_rate_alert

File: logdna/request_types.go
Comment (seeruyy) [reply logdna/request_types.go:198]: That's ok, nothing unusual, `make test` and `make test-local` run `go fmt` by default.

File: docs/resources/logdna_index_rate_alert.md
Comment (seeruyy) [reply docs/resources/logdna_index_rate_alert.md:27]: Also, I'd update this file as well.
https://github.com/logdna/terraform-provider-logdna/blob/main/examples/index_rate_alert.tf

File: logdna/request_types.go
Comment (jsumners) [reply logdna/request_types.go:198]: @seeruyy it gets corrected by `go fmt`. I ran it against this branch's file before commenting 😉

File: logdna/response_types.go
Comment (jsumners) [reply logdna/response_types.go:116]: FYI Arani: `ConvertWebhook` is a public method whereas `convertWebhook` is a package private one.

File: logdna/provider.go
Comment (arani-sen) [reply logdna/provider.go:27]: No, i thought I ignored this file.

**Comment by arani-sen** [comment]
> I believe we should get #88 merged first so that you can run `go fmt ./...` on your branch and see minimal changes. Or, we can ignore my comments about formatting and I'll rebase my PR after this one has merged.
> 
> But before either can be merged we need @jakedipity to answer a question I have put to him in Slack about the failing tests.

I wanted to wait for your merge first so that i can run my format as well.

File: logdna/response_types.go
Comment (arani-sen) [reply logdna/response_types.go:116]: I thought we were talking about having this part of the Indexrate alert response. i will make the change

File: logdna/request_types.go
Comment (seeruyy) [reply logdna/request_types.go:198]: I see your PR, yeah, that makes sense.

**Comment by seeruyy** [comment]
It looks good to me now, but we still have an issue with Key API test which is not related to this PR.
@jakedipity Could you please assist with this issue?

**Review by seeruyy — DISMISSED** [review]

**Comment by arani-sen** [comment]
> It looks good to me now, but we still have an issue with Key API test which is not related to this PR. @jakedipity Could you please assist with this issue?

This fix is related to the issue https://github.com/answerbook/logdna-api/pull/825

File: logdna/request_types.go
Comment (jsumners) [reply logdna/request_types.go:198]: Also, I'm going to look into what we need to add to get some checks on this going forward.

Comment (arani-sen) [reply logdna/request_types.go]: formatting added now

**Review by jsumners — DISMISSED** [review]
Looks good to me.

**Review by jsumners — APPROVED** [review]

**Pull request merged**
